jupyter_server/gateway/gateway_client.py

```diff
@@ -808,6 +811,16 @@ async def gateway_request(endpoint: str, **kwargs: ty.Any) -> HTTPResponse:
             f"The Gateway server specified in the gateway_url '{GatewayClient.instance().url}' doesn't "
             f"appear to be valid.  Ensure gateway url is valid and the Gateway instance is running.",
         ) from e
+    except Exception as e:
+        GatewayClient.instance().emit(
+            data={STATUS_KEY: ERROR_STATUS, STATUS_CODE_KEY: 505, MESSAGE_KEY: str(e)}
+        )
+        logger = logging.getLogger(__name__)
```

Comment:
Maybe use the same pattern as above -> https://github.com/jupyter-server/jupyter_server/blob/6b9a86efc8c59b8e6348bb6403b51cf141b1b732/jupyter_server/gateway/gateway_client.py#L735

Reply:
Updated..